### PR TITLE
Merge `I32BasedType` & `U32BasedType` together into `NumberBasedType`

### PR DESCRIPTION
**Pull request by FirelightFlagboy**

### CODE REVIEW

**Review by TimeEngineer — CHANGES_REQUESTED** [review]
We don't want that, see: https://github.com/Scille/parsec-cloud/issues/3941

**Comment by FirelightFlagboy** [comment]
I don't understand what's blocking can you elaborate ?

**Comment by TimeEngineer** [comment]
> I don't understand what's blocking can you elaborate ?

The changes are in contradiction with the issue

File: bindings/generator/templates/binding_electron_meths.rs.j2 [hunk 1]
```diff
@@ -276,7 +269,7 @@ match {{ rs_value }} {
 }
 {%- elif type.kind == "bool" -%}
 JsBoolean::new({{ mut_cx_ref }}, {{ rs_value }})
-{%- elif type.kind in ("i32_based", "u32_based") -%}
+{%- elif type.kind in ("number_based") -%}
 JsNumber::new({{ mut_cx_ref }}, {{ rs_value }} as f64)
```
Comment (TimeEngineer) [inline bindings/generator/templates/binding_electron_meths.rs.j2:273]: This number can be invalid for `u64` or `i64`

**Comment by FirelightFlagboy** [comment]
Outdated, closing

**Pull request closed without merging**